sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -211,6 +211,77 @@ case class EndsWith(left: Expression, right: Expression)
 }
 
 /**
+ * A function that returns the index (1-based) of the given string (left) in the comma-
+ * delimited list (right). Returns -1, if the string wasn't found and 0 if the given
+ * string (left) contains a comma.
+ */
+case class FindInSet(left: Expression, right: Expression) extends BinaryExpression
+    with ExpectsInputTypes {
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    (left.dataType, right.dataType) match {
+      case (_, NullType) | (NullType, _) | (StringType, StringType) =>
+        TypeCheckResult.TypeCheckSuccess
+      case _ =>
+        TypeCheckResult.TypeCheckFailure(s"FindInSet expects two strings as argument, not " +
+          s"(${left.dataType}, ${right.dataType})")
+    }
+  }
+
+  override def inputTypes: Seq[Any] = Seq(StringType, StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val valueLeft = left.eval(input)
+    if (valueLeft != null) {
+      val strLeft = valueLeft.asInstanceOf[UTF8String]
+      if (!strLeft.contains(UTF8String.fromString(","))) {
+        val valueRight = right.eval(input)
+        if (valueRight != null) {
+          val splits = valueRight.asInstanceOf[UTF8String].toString.split(",")
+          var i = 0
```

Comment:
I think the code has performance issue. When you call split, it scans the string for one time. After that, you process the string for the second time. If the set is large, the performance issue will appear. You could scan the set by one time.